### PR TITLE
implemented authentication and added unit tests

### PR DESCRIPTION
**Pull request by dermo666**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -8,7 +9,12 @@ app.get('/status', (req, res) => { res.status(200).end(); });
 app.head('/status', (req, res) => { res.status(200).end(); });
 
 const basicAuthHandler = (req: express.Request, res: express.Response, next: express.NextFunction) => {
-  if (true) {
+  const b64auth = (req.headers.authorization || '').split(' ')[1] || '';
```
Comment (vince86400) [inline src/index.ts:12]: Hi Tomas thanks for submitting this! There are potentially some edge cases not being handled.
Consider that the Basic Authentication scheme expects the following format for the value:
`Basic: credentials`
The auth method has to start with the word `Basic:`. Credentials is a base64 encoded value of a string in the format of _username:password_ This would mean at a minimum username and password would not be able to contain the `:` character as well.
Based on the above would you be able to add some test scenarios and code to cover these edge cases?

Comment (dermo666) [reply src/index.ts:12]: Hi, I have already added one scenario with invalid `credentials` : 
`.set('Authorization', 'Basic: dummy');`

I can add some extra ones eg. without the `Basic` keyword etc.

Comment (dermo666) [reply src/index.ts:12]: I added two more cases:
* when username contains a colon -> this should be considered as limitation of the Basic Auth and handled as registration to now allow emails with colon
* when password contains a colon -> this has been handled in the code and is allowed
